src/filter.rs

```diff
@@ -76,3 +76,181 @@ impl Filter {
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+
+    // TODO Brian: Find a unique access point for the following fn's, repeated in messages.rs.
+    fn generate_random_u8() -> u8 {
+        use rand::Rng;
+
+        let mut rng = ::rand::thread_rng();
+        rng.gen::<u8>()
+    }
+
+    fn generate_random_vec() -> ::std::vec::Vec<u8> {
```

Comment:
Random functions are only used for testing so that function as well as the new ones should probably be written to test_utils and made pubclicly available to tests. Perhaps another task for that would be appropriate.